cpp/include/table.hpp

```diff
@@ -33,6 +34,13 @@ namespace cudf {
  *
  */
 struct table {
+  /**---------------------------------------------------------------------------*
```

Comment:
In my current PR, I had added a new constructor that takes a `std::vector<gdf_dtype>` to enable creating an empty table of the same types as another table (output table given input table). But I like your table empty_like approach better...